syntax = "proto3";

import "mlagents_envs/communicator_objects/agent_action.proto";
import "mlagents_envs/communicator_objects/command.proto";

option csharp_namespace = "MLAgents.CommunicatorObjects";
package communicator_objects;

message UnityRLInputProto {
    message ListAgentActionProto {
            repeated AgentActionProto value = 1;
        }
    map<string, ListAgentActionProto> agent_actions = 1;
    reserved 2; //deprecated environment proto
    reserved 3; //deprecated is_trainig
    CommandProto command = 4;
    bytes side_channel = 5;
}
